lib/rack/deflater.rb

```diff
@@ -66,7 +66,13 @@ def call(env)
         headers['content-encoding'] = "gzip"
         headers.delete(CONTENT_LENGTH)
         mtime = headers["last-modified"]
-        mtime = Time.httpdate(mtime).to_i if mtime
+        if mtime
```

Comment:
Is this conditional could be a new method or helper module?

```rb
module TimeHelper
  def httpdate(mtime = nil)
    return if mtime.nil?

    Time.httpdate(mtime) rescue nil
  end
end

mtime = TimeHelper.httpdate(headers['last-modified'])
```

Reply:
I'm generally against unnecessary indirection, so I do not think this would improve things.